prov/rxd/src/rxd_ep.c

```diff
@@ -1011,8 +1011,11 @@ void rxd_ep_progress(struct util_ep *util_ep)
 	}
 
 out:
-	while (ep->posted_bufs < ep->rx_size && !ret)
+	while (ep->posted_bufs < ep->rx_size) {
 		ret = rxd_ep_post_buf(ep);
+		if (ret)
+			break;
+	}
```

Comment:
This change is fine, but please look at changing the code structure so that the variables that control the while loop are visibly updated inside the loop, rather than as a by-product hidden inside some other call.  Consider moving the while loop into the rxd_ep_post_buf call, for example.